icu4j/main/classes/core/src/com/ibm/icu/impl/units/UnitConverter.java

```diff
@@ -0,0 +1,315 @@
+    }
+
+    public enum Convertibility {
+        CONVERTIBLE,
+        RECIPROCAL,
+        INCONVERTIBLE,
+    }
+
+    /**
+     * Responsible for all the Factor operation
+     * NOTE:
+     * This class is immutable
+     */
+    static class Factor {
+        private BigDecimal factorNum;
+        private BigDecimal factorDen;
+        /* FACTOR CONSTANTS */
+        private int
+                CONSTANT_FT2M = 0,    // ft2m stands for foot to meter.
+                CONSTANT_PI = 0,      // PI
+                CONSTANT_GRAVITY = 0, // Gravity
+                CONSTANT_G = 0,
+                CONSTANT_GAL_IMP2M3 = 0, // Gallon imp to m3
+                CONSTANT_LB2KG = 0;      // Pound to Kilogram
+
+
+        /**
+         * Creates Empty Factor
+         */
+        public Factor() {
+            this.factorNum = BigDecimal.valueOf(1);
+            this.factorDen = BigDecimal.valueOf(1);
+        }
+
+        public static Factor precessFactor(String factor) {
+            Assert.assrt(!factor.isEmpty());
+
+            // Remove all spaces in the factor
+            factor.replaceAll("\\s+", "");
+
+            String[] fractions = factor.split("/");
+            Assert.assrt(fractions.length == 1 || fractions.length == 2);
+
+            if (fractions.length == 1) {
+                return processFactorWithoutDivision(fractions[0]);
+            }
+
+            Factor num = processFactorWithoutDivision(fractions[0]);
+            Factor den = processFactorWithoutDivision(fractions[1]);
+            return num.divide(den);
+        }
+
+        private static Factor processFactorWithoutDivision(String factorWithoutDivision) {
+            Factor result = new Factor();
+            for (String poweredEntity :
+                    factorWithoutDivision.split(Pattern.quote("*"))) {
+                result.addPoweredEntity(poweredEntity);
+            }
+
+            return result;
+        }
+
+        /**
+         * Clone this <code>Factor</code>.
+         */
+        protected Factor clone() {
+            Factor result = new Factor();
+            result.factorNum = this.factorNum;
+            result.factorDen = this.factorDen;
+
+            result.CONSTANT_FT2M = this.CONSTANT_FT2M;
+            result.CONSTANT_PI = this.CONSTANT_PI;
+            result.CONSTANT_GRAVITY = this.CONSTANT_GRAVITY;
+            result.CONSTANT_G = this.CONSTANT_G;
+            result.CONSTANT_GAL_IMP2M3 = this.CONSTANT_GAL_IMP2M3;
+            result.CONSTANT_LB2KG = this.CONSTANT_LB2KG;
+
+            return result;
+        }
+
+        /**
+         * Returns a single `BigDecimal` that represent the conversion rate after substituting all the constants.
+         *
+         * @return
+         */
+        public BigDecimal getConversionRate() {
+            Factor resultCollector = this.clone();
+
+            resultCollector.substitute(BigDecimal.valueOf(0.3048), this.CONSTANT_FT2M);
```

Comment:
Here and below: contrary to intuition, `new BigDecimal("0.3048")` is better than `BigDecimal.valueOf(0.3048)` because the string is more exact